packages/@react-aria/datepicker/src/useDatePicker.ts

```diff
@@ -76,7 +76,63 @@ export function useDatePicker<T extends DateValue>(props: AriaDatePickerProps<T>
       role: 'group',
       'aria-disabled': props.isDisabled || null,
       'aria-labelledby': labelledBy,
-      'aria-describedby': ariaDescribedBy
+      'aria-describedby': ariaDescribedBy,
+      onKeyDown(e: KeyboardEvent) {
+        if (state.isOpen) {
+          return;
+        }
+
+        if (props.onKeyDown) {
+          props.onKeyDown(e);
+        }
+      },
+      onKeyUp(e: KeyboardEvent) {
+        if (state.isOpen) {
+          return;
+        }
+
+        if (props.onKeyUp) {
+          props.onKeyUp(e);
+        }
+      },
+      onFocus(e: ReactFocusEvent) {
+        if (state.isFocused) {
```

Comment:
This was modeled on how Picker handles these events. Picker had the `state.isFocused` and `e.currentTarget.contains(e.relatedTarget as Node)` logic. In this case, the `state.isFocused` is to not keep calling focus when moving between the segments and the trigger button. The `e.currentTarget.contains(e.relatedTarget as Node)` keeps the blur from being called when moving between segments and to the trigger button. The `state.isOpen` keeps the focus, blur, and keyup/down events from being called while using the calendar popover. There weren't comments on this in Picker so I'm not sure if we want them here.

While this logic is identical for the two picker hooks, date and time fields don't have a popover and `state.isOpen`, so it didn't seem worthwhile to generalize.